CONTRIBUTING_CODE.md

```diff
@@ -0,0 +1,302 @@
+Next, set up a branch on your fork which is tracked by your local branch. 
+```shell
+git push --set-upstream origin my-new-feature
+```
+Once in a while, merge changes of the upstream master branch into your local `my-new-feature` branch.
+```shell
+$ git checkout probnum-master
+$ git pull
+$ git checkout my-new-feature
+$ git merge probnum-master
+```
+### Open a pull-request
+Once you are happy with your PR, make sure your branch is up-to-date with upstream `master`, and re-run all tests.
+Then, from your local branch, push the changes to your remote branch
+```shell
+$ git push
+```
+
+and do the PR via the github interface. There, briefly explain the changes as well as how they were tested. That's it!
+
+
+## Virtual environments
+
+Virtual environments (venvs) help you to get a separate, clean installation of ProbNum. 
+Each venv uses a specific Python version.
+Venvs are useful as you may need different Python versions and dependencies for different projects.
+So, first make sure that you are using the correct Python version, and also that your pip 
+installation is up to date.
+
+Install the [virtualenv](https://virtualenv.pypa.io/en/latest/) package.
+```shell
+$ pip install virtualenv
+```
+
+Go to the probnum root directory. 
+Then, create a virtual environment wit the name `venv_probnum`. 
+This uses the Python version the `python` alias is linked to. 
+```shell
+$ python -m venv venv_probnum
+```
+
+
+Activate the venv (the command below works for bash, other shells might require a different command).
+```shell
+$ source venv_probnum/bin/activate
+```
+
+Check if your venv uses the correct Python path and Python version.
+(The Python path should show something like this `/home/MyUserName/probnum/venv_probnum/bin/python`.)
+```shell
+$ which python
+$ python --version
+```
+
+
+You can deactivate the venv as well.
+```shell
+$ deactivate
+```
+
+
+
+Install probnum in your environment (activate it first). 
+```shell
+$ pip install -e .
+```
+
+If needed, install additional requirements, e.g., for testing.
+```shell
+$ pip install -e .[test]
+```
+
+Check your installation with `pip freeze`.
+
+
+It is possible to create virtual environments with most modern editors such as e.g., PyCharm. Alternatively, you can 
+add the venv you created above to your project in the editor. 
+
+## Testing
+Tests are run with the [pytest package](https://docs.pytest.org/en/stable/). 
+Make sure ProbNum is installed including test requirements in your virtual environment; 
+installing ProbNum that way will already have installed pytest.
+(If you are unsure, use `pip freeze` to check the installation in your active venv). 
+
+Run tests with the Python version in the venv.
+```shell
+$ pytest
+```
+This should normally be enough to catch the biggest bugs. If you want to run the whole testsuite for several Python 
+versions, the preferred way is to use tox.
+
+
+## tox
```

Comment:
title case

Reply:
It's a brand name, and they have it lower case on their own [page](https://github.com/tox-dev/tox) as well, so I think it's better to keep it lower case.